Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerFileImpl.cs

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+using Mono.Unix;
+using Mono.Unix.Native;
+using Newtonsoft.Json;
+using Snowflake.Data.Client;
+using Snowflake.Data.Core.Tools;
+using Snowflake.Data.Log;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using KeyToken = System.Collections.Generic.Dictionary<string, string>;
+
+namespace Snowflake.Data.Core.CredentialManager.Infrastructure
+{
+    internal class SFCredentialManagerFileImpl : ISnowflakeCredentialManager
```

Comment:
I'm pretty sure we've already had a talk about it, but I think there is still a misunderstanding...
We want the user to be able to change which implementation he wants to use so he has a method `SetCredentialManager` for that.
But to be able to do it, the implementation class need to be public.
And if it is public it should start from Snowflake not SF and be in the Client package not Core.

Reply:
no, it was a comment for all 3 implementations - I think they should be treated in the same way.

But when I'm thinking about it now, maybe let's do it differently.
Actually let's make all 3 implementations as internal (so the one moved to client should be moved back).
Then we could add to SnowflakeCredentialManagerFactory 3 public properties having instance to these 3 implementations. The properties would be only with getters (no setters, readonly).
Thanks to that these classes could be internal.

Reply:
I moved back the SFFileImplCredentialManager back to the Core. Instead of having properties for each 3 implementations to use the SetCredentialManager function I will maintained internal and create 3 methods to enable the defaults implemented credentials.